src/transformers/models/higgs_audio_v2_tokenizer/modular_higgs_audio_v2_tokenizer.py

```diff
@@ -0,0 +1,169 @@
+                Number of entries in each residual quantizer's codebook.
+            codebook_dim (`int`, *optional*):
+                Dimensionality of each codebook vector. Defaults to sum of hidden size of acoustic and semantic models.
+            initializer_range (`float`, *optional*, defaults to 0.02):
+                Standard deviation of the truncated normal initializer for all weight matrices.
+            acoustic_model_config (`Union[Dict, DacConfig]`, *optional*):
+                An instance of the configuration for the acoustic (DAC) model.
+            semantic_model_config (`Union[Dict, HubertConfig, WavLMConfig]`, *optional*):
+                An instance of the configuration object for the semantic (HuBERT) model.
+
+    Example:
+
+    ```python
+    >>> from transformers import HiggsAudioV2TokenizerModel, HiggsAudioV2TokenizerConfig
+
+    >>> # Initializing configuration
+    >>> configuration = HiggsAudioV2TokenizerConfig()
+
+    >>> # Initializing a model (with random weights) from the configuration
+    >>> model = HiggsAudioV2TokenizerModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    def __init__(
+        self,
+        target_bandwidths=None,
+        sample_rate=24000,
+        kernel_size=3,
+        channel_ratios=[1, 1],
+        strides=[1, 1],
+        block_dilations=[1, 1],
+        unit_kernel_size=3,
+        codebook_size=1024,
+        codebook_dim=None,
+        initializer_range=0.02,
+        acoustic_model_config=None,
+        semantic_model_config=None,
+        semantic_sample_rate=16000,
+        downsample_factor=320,
+        **kwargs,
+    ):
+        super().__init__(
+            target_bandwidths=target_bandwidths,
+            sample_rate=sample_rate,
+            kernel_size=kernel_size,
+            channel_ratios=channel_ratios,
+            strides=strides,
+            block_dilations=block_dilations,
+            unit_kernel_size=unit_kernel_size,
+            codebook_size=codebook_size,
+            codebook_dim=codebook_dim,
+            initializer_range=initializer_range,
+            acoustic_model_config=acoustic_model_config,
+            semantic_model_config=semantic_model_config,
+            **kwargs,
+        )
+
+        self.semantic_sample_rate = semantic_sample_rate
+        self.downsample_factor = downsample_factor
+
+
+class HiggsAudioV2TokenizerEuclideanCodebook(XcodecEuclideanCodebook): ...
+
+
+class HiggsAudioV2TokenizerVectorQuantization(nn.Module):
+    def __init__(self, config: HiggsAudioV2TokenizerConfig):
+        super().__init__()
+        self.codebook = HiggsAudioV2TokenizerEuclideanCodebook(config)
+        codebook_dim = config.codebook_dim
+        dim = config.hidden_size
+        requires_projection = codebook_dim != dim
+        self.project_in = nn.Linear(dim, codebook_dim) if requires_projection else nn.Identity()
```

Comment:
are there 2 differrent models where one requires projection and the other does not? IF not let's remove the if elses